sycl-jit/common/include/DynArray.h

```diff
@@ -22,6 +23,11 @@ template <typename T> class DynArray {
 
   explicit DynArray(size_t Size) { init(Size); }
 
+  explicit DynArray(size_t Size, const void *Data) {
+    init(Size);
+    memcpy(static_cast<void *>(this->begin()), Data, Size * sizeof(T));
+  }
```

Comment:
```suggestion
  template <typename InputIt>
  DynArray(InputIt begin, InputIt end) {
    init(end - begin);
    std::copy(begin, end, this->begin());
  }
```

What about this?

Reply:
Sure, if you prefer iterator approach, I'm cool with that. Done.